mlbench_core/utils/pytorch/__init__.py

```diff
@@ -7,12 +7,15 @@
 
 from contextlib import contextmanager
 
+import os
 __all__ = ["initialize_backends", "Timeit", "FCGraph"]
 
 
 @contextmanager
 def initialize_backends(
```

Comment:
Maybe we could add a check for CUDA devices when using NCCL backend. You can check whether `torch.cuda.is_available` when using the NCCL backend, and if it's not available, raise a coherent error message